source/objects/cableClass.m

```diff
@@ -0,0 +1,283 @@
+            end
+            x = cross(y,z)/norm(cross(y,z));
+            x = x(:)';
+            obj.orientation.x = x;
+            obj.orientation.rotationMatrix  = [x',y',z'];
+            %
+            % This method calculates the constraint ``x`` vector and ``rotationMatrix`` matrix in the ``orientation`` structure based on user input.
+            obj.rotorientation.z = obj.rotorientation.z / norm(obj.rotorientation.z);
+            obj.rotorientation.y = obj.rotorientation.y / norm(obj.rotorientation.y);
+            z = obj.rotorientation.z;
+            y = obj.rotorientation.y;
+            if abs(dot(y,z))>0.001
+                error('The Y and Z vectors defining the constraint''s orientation must be orthogonal.')
+            end
+            x = cross(y,z)/norm(cross(y,z));
+            x = x(:)';
+            obj.rotorientation.x = x;
+            obj.rotorientation.rotationMatrix1  = [x',y',z'];
+            obj.rotorientation.rotationMatrix2  = [x',y',z'];
+        end
+        
+        function setInitDisp(obj, x_rot, ax_rot, ang_rot, addLinDisp)
+            % This method sets initial displacement while considering an initial rotation orientation.
+            %
+            %``x_rot`` (`3x1 float vector`) is rotation point [m] in the following format [x y z], Default = ``[]``.
+            %
+            %``ax_rot`` (`3x1 float vector`) is the axis about which to rotate to constraint and must be a normal vector, Default = ``[]``.
+            %
+            %``ang_rot`` (`float`) is the rotation angle [rad], Default = ``[]``.
+            %
+            %``addLinDisp`` ('float') is the initial linear displacement [m] in addition to the displacement caused by the pto rotation, Default = '[]'.
+            loc = obj.loc;
+            relCoord = loc - x_rot;
+            rotatedRelCoord = rotateXYZ(relCoord,ax_rot,ang_rot);
+            newCoord = rotatedRelCoord + x_rot;
+            linDisp = newCoord-loc;
+            obj.initDisp.initLinDisp= linDisp + addLinDisp;
+            %
+            rotloc1 = obj.rotloc1;
+            relCoord = rotloc1 - x_rot;
+            rotatedRelCoord = rotateXYZ(relCoord,ax_rot,ang_rot);
+            newCoord = rotatedRelCoord + x_rot;
+            linDisp = newCoord-rotloc1;
+            obj.rotinitDisp.initLinDisp1= linDisp + addLinDisp;
+            %
+            rotloc2 = obj.rotloc2;
+            relCoord = rotloc2 - x_rot;
+            rotatedRelCoord = rotateXYZ(relCoord,ax_rot,ang_rot);
+            newCoord = rotatedRelCoord + x_rot;
+            linDisp = newCoord-rotloc2;
+            obj.rotinitDisp.initLinDisp2= linDisp + addLinDisp;
+        end
+        
+        function checkFloat(obj, rho)
+            dispVol = obj.dispVol;
+            dispMass = obj.dispVol * rho;
+            if dispMass ~= obj.bodyMass;
+                warning('Cable dummy bodies are not neutrally buoyant. Check simu.rho and cable.bodyMass')
+            end
+        end
```

Comment:
Unless there's a good reason to allow the dummy bodies not to be neutrally buoyant, make dispVol private. Define the volume in dragprecalc as body.mass/simu.rho to enforce neutral buoyancy

Reply:
Treating this like a low-order lumped capacitance model of the cable, there is a good reason to retain this functionality. I think 90% of users will want it neutrally buoyant, so I've made this the default, but I prefer to not bury the property.

Reply:
Okay this makes sense, let's keep it this way